File: 백준/Silver/20125. 쿠키의 신체 측정/쿠키의 신체 측정.py
import sys
input = sys.stdin.readline

n = int(input())

arr = [input().rstrip() for _ in range(n)]

heart = [0, 0]
flag = False
for i in range(n):
    for j in range(n):
        if arr[i][j] == '*':
            heart[0], heart[1] = i + 1, j
            flag = True
            break
    if flag:
        break

left_arm, right_arm, waist, left_leg, right_leg = 0, 0, 0, 0, 0


def check(r, c):
    return 0 <= r < n and 0 <= c < n


r, c = heart[0], heart[1] - 1
while check(r, c) and arr[r][c] == '*':
    left_arm += 1
    c -= 1

c = heart[1] + 1
while check(r, c) and arr[r][c] == '*':
    right_arm += 1
    c += 1

r, c = heart[0] + 1, heart[1]
while check(r, c) and arr[r][c] == '*':
    waist += 1
    r += 1

low = r
c = heart[1] - 1
while check(r, c) and arr[r][c] == '*':
    left_leg += 1
    r += 1

r = low
c = heart[1] + 1
while check(r, c) and arr[r][c] == '*':
    right_leg += 1
    r += 1

print(heart[0] + 1, heart[1] + 1)
print(left_arm, right_arm, waist, left_leg, right_leg)

File: 백준/Silver/20125. 쿠키의 신체 측정/쿠키의 신체 측정.java
import java.io.BufferedReader;
import java.io.InputStreamReader;

public class Main {
    static int N;
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        N = Integer.parseInt(br.readLine());

        char[][] arr = new char[N][N];
        boolean flag = false;
        int[] heart = new int[2];
        for(int i = 0; i < N; i++) {
            String s = br.readLine();
            for(int j = 0; j < N; j++) {
                arr[i][j] = s.charAt(j);
                if(!flag && arr[i][j] == '*') {
                    flag = true;
                    heart[0] = i + 1; heart[1] = j;
                }
            }
        }

        StringBuilder sb = new StringBuilder();
        sb.append(heart[0] + 1).append(" ").append(heart[1] + 1).append("\n");

        int leftArm = 0, rightArm = 0, waist = 0, leftLeg = 0, rightLeg = 0;
        int low;
        int nr = heart[0], nc = heart[1] - 1;
        while(check(nr, nc) && arr[nr][nc] == '*') {
            leftArm++;
            nc--;
        }
        nc = heart[1] + 1;
        while(check(nr, nc) && arr[nr][nc] == '*') {
            rightArm++;
            nc++;
        }
        nr = heart[0] + 1; nc = heart[1];
        while(check(nr, nc) && arr[nr][nc] == '*') {
            waist++;
            nr++;
        }
        low = nr;
        nc = heart[1] - 1;
        while(check(nr, nc) && arr[nr][nc] == '*') {
            leftLeg++;
            nr++;
        }
        nc = heart[1] + 1;
        nr = low;
        while(check(nr, nc) && arr[nr][nc] == '*') {
            rightLeg++;
            nr++;
        }

        sb.append(leftArm).append(" ").append(rightArm).append(" ").append(waist).append(" ").append(leftLeg).append(" ").append(rightLeg);
        System.out.println(sb);
        br.close();
    }

    public static boolean check(int r, int c) {
        return 0 <= r && r < N && 0 <= c && c < N;
    }
}
